modules/react/pagination/stories/examples/ResponsiveRange.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import {
+  Pagination,
+  getLastPage,
+  getVisibleResultsMax,
+  getVisibleResultsMin,
+  usePaginationModel,
+} from '@workday/canvas-kit-react/pagination';
+import {Flex} from '@workday/canvas-kit-react/layout';
+import {Text} from '@workday/canvas-kit-react/text';
+import {useResizeObserver, useTheme} from '@workday/canvas-kit-react/common';
+
+export const ResponsiveRange = () => {
+  const resultCount = 10;
+  const totalCount = 100;
+  const lastPage = getLastPage(resultCount, totalCount);
+  // Normally, rangeSize can be a static value, but we're making it stateful,
+  // so we can update it using useResizeObserver
+  const [rangeSize, setRangeSize] = React.useState(5);
+  const model = usePaginationModel({
+    lastPage,
+    rangeSize,
+  });
+  // We're only using this state to display the container width;
+  const [containerWidth, setContainerWidth] = React.useState(0);
+  // We're using breakpoints from the theme as reference points to adjust the range
+  const theme = useTheme();
+  const {values: breakpointValues} = theme.canvas.breakpoints;
+
+  // We'll use this ref to measure the size of the container
+  const containerRef = React.useRef(null);
+  useResizeObserver({
+    ref: containerRef,
+    onResize: ({width}) => {
+      // Note: onResizeObserver only accounts for the size of the container.
+      // It does not factor in margin, padding, or border widths.
+      // So, if you want to be exact, you'll need to adjust your math a bit to account for those.
```

Comment:
Optional word removal:
```suggestion
      // If you want to be exact, adjust your math to account for those.
```